### ISSUE TITLE
Model checkpoint access

### ISSUE THREAD

**Issue by kyoyachuan** [DianJin-HFUT/SimToken#3]
Congrats on the ICASSP 2026 acceptance, great work! 
I’d like to run the evaluation, but the checkpoint link seems to be access-restricted. Any chance you could make it available? 
Thanks a lot!

**Comment by kyoyachuan** [comment]
Thanks for sharing the checkpoint! I’ve tested the model, but due to my 24 GB GPU memory limit, I had to run it with bfloat16 instead of FP32. I noticed a substantial performance drop, i.e., J&F on the Seen subset is 43.9 compared to 76.7 in the paper. I’m not sure if this is expected when using bfloat16, or if there are specific inference settings you’d recommend. Any guidance would be appreciated.

**Comment by kyoyachuan** [comment]
Thanks for the follow-up! Here are the details of my setup:
- I loaded the model directly with torch_dtype=torch.bfloat16 (not casting afterward with .to()).
- I did not explicitly convert individual modules to bfloat16.
- I used AMP / torch.autocast during inference.
- I did not manually convert the vision encoder or mask decoder to bfloat16.

The exact code changes are as follow

```python
model = Simtoken_ForCausalLM.from_pretrained(args.mllm, torch_dtype=torch.bfloat16, **model_args)
```

```python
        for batch in tqdm(dataloader, desc=f"Evaluating on {name}"):
            input_dict = dict_to_cuda(batch)
            with torch.amp.autocast(device_type="cuda", dtype=torch.bfloat16, enabled=args.bf16):
                with torch.no_grad():
                    output_dict = model.forward(images=input_dict["images"],
                                                images_clip=input_dict["images_clip"],
                                                audio_features=input_dict["audio_feats"],
                                                image_features=input_dict["image_feats"],
                                                input_ids=input_dict["input_ids"],
                                                labels=input_dict["labels"],
                                                attention_masks=input_dict["attention_masks"],
                                                masks_list=input_dict["masks"],
                                                resize_list=input_dict["resizes"],
                                                orgsize_list=input_dict["orgsizes"],
                                                conversation_list=input_dict["convs"],
                                                refs_num=input_dict["refs_num"],
                                                fids=input_dict["fids"],
                                                vids=input_dict["vids"],
                                                contrast=args.ct_weight,
                                                ref_ids=input_dict["ref_ids"],
                                                inference=True)
```

**Comment by DianJin-HFUT** [comment]
Thanks again for the detailed experiments and for your continued engagement.

To further investigate this, we also ran additional tests on our side using a different platform from our original training environment. Specifically, we set up a 24 GB RTX 4090 instance on AutoDL, rebuilt the environment from scratch, re-extracted the features, and evaluated the released checkpoint using bf16 precision.

With this setup, we obtained a J&F score of approximately 75.4, which is very close to the results reported in the paper and consistent with our original platform. This suggests that, in our experiments, bf16 inference itself does not lead to a significant performance degradation.

Based on these results, we are currently unable to reproduce the substantial drop you observed. This indicates that the issue may be related to additional environment- or configuration-specific factors beyond model precision alone (e.g., library versions, CUDA / PyTorch behavior, or inference settings).

At this point, we don’t yet have a clear explanation for the discrepancy, but we’re actively looking into it on our side as well.
If you come across any new observations or have questions about particular settings or configurations, please don’t hesitate to let us know.
